Apps/Admin/Client/Components/Support/Covid19ImmunizationSection.razor.cs

```diff
@@ -0,0 +1,193 @@
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Admin.Client.Components.Support
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Fluxor;
+    using Fluxor.Blazor.Web.Components;
+    using HealthGateway.Admin.Client.Store.VaccineCard;
+    using HealthGateway.Admin.Common.Models.CovidSupport;
+    using HealthGateway.Common.Data.Models;
+    using HealthGateway.Common.Data.Utils;
+    using Microsoft.AspNetCore.Components;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.JSInterop;
+    using MudBlazor;
+    using MailVaccineCardAddressConfirmationDialog = AddressConfirmationDialog<
+        Store.VaccineCard.VaccineCardActions.MailVaccineCardFailureAction,
+        Store.VaccineCard.VaccineCardActions.MailVaccineCardSuccessAction>;
+
+    /// <summary>
+    /// The covid19 immunization section component.
+    /// </summary>
+    public partial class Covid19ImmunizationSection : FluxorComponent
+    {
+        /// <summary>
+        /// Gets or sets the patient's phn.
+        /// </summary>
+        [Parameter]
+        [EditorRequired]
+        public string Phn { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the data with which to populate the table.
+        /// </summary>
+        [Parameter]
+        [EditorRequired]
+        public IEnumerable<VaccineDose> Data { get; set; } = Enumerable.Empty<VaccineDose>();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether data is loading.
+        /// </summary>
+        [Parameter]
+        [EditorRequired]
+        public bool IsLoading { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the patient's address.
+        /// </summary>
+        [Parameter]
+        [EditorRequired]
+        public Address? MailAddress { get; set; }
+
+        [Inject]
+        private IActionSubscriber ActionSubscriber { get; set; } = default!;
+
+        [Inject]
+        private IConfiguration Configuration { get; set; } = default!;
+
+        [Inject]
+        private IDialogService Dialog { get; set; } = default!;
+
+        [Inject]
+        private IDispatcher Dispatcher { get; set; } = default!;
+
+        [Inject]
+        private IJSRuntime JsRuntime { get; set; } = default!;
+
+        [Inject]
+        private IState<VaccineCardState> VaccineCardState { get; set; } = default!;
+
+        private bool MailVaccineCardIsLoading => this.VaccineCardState.Value.MailVaccineCard.IsLoading;
+
+        private bool PrintVaccineCardIsLoading => this.VaccineCardState.Value.PrintVaccineCard.IsLoading;
+
+        private IEnumerable<VaccineDoseRow> Rows => this.Data.Select(mv => new VaccineDoseRow(mv));
+
+        private ReportModel VaccineCardStateData => this.VaccineCardState.Value.PrintVaccineCard.Result ?? default!;
```

Comment:
I suggest making this nullable since we are not guaranteed to have a result.

Reply:
Changed.